components/form.tsx

```diff
@@ -0,0 +1,120 @@
+import React, { useState, ChangeEvent, FormEvent, FC } from 'react';
+import styled from 'styled-components';
+import Heading1 from '../elements/Heading1';
+import Heading2 from '../elements/Heading3';
+import Heading3 from '../elements/Heading3';
+import Paragraph1 from '../elements/Paragraph1';
+import SubmitButton from '../elements/SubmitButton';
+import FormElement from '../elements/FormElement';
+import TextInput from '../elements/TextInput';
+import { postTestimonial } from '../services/airTableService';
+import testimonial from '../interfaces/testimonial';
+
+// const FormElement = styled.form`
+//   margin: 2rem 2rem;
+//   width: 80%;
+//   display: flex;
+//   flex-direction: column;
+//   justify-content: center;
+//   align-items: center;
+//   box-sizing: border-box;
+//   box-shadow: rgba(50, 50, 93, 0.25) 0px 50px 100px -20px,
+//     rgba(0, 0, 0, 0.3) 0px 30px 60px -30px;
+//   border-radius: 1rem;
+// `;
+
+const Form: FC = () => {
+  const emptyForm: testimonial = {
+    testimony: '',
+    location: '',
+    consent: false,
+  };
+  const [testimonial, setTestimonial] = useState(emptyForm);
+
+  const handleChange = (
+    e: ChangeEvent<HTMLInputElement | HTMLTextAreaElement>
+  ): void => {
+    const name: string = e.target.name;
+    const value: string | boolean =
+      e.target.type === 'checkbox' ? !testimonial.consent : e.target.value;
+
+    setTestimonial(prevTestimonial => ({
+      ...prevTestimonial,
+      [name]: value,
+    }));
+  };
+
+  const handleSubmit = async (e: FormEvent<HTMLFormElement>): Promise<void> => {
+    e.preventDefault();
+    await postTestimonial(testimonial);
+    setTestimonial(emptyForm);
+  };
+
+  return (
+    <FormElement m={7} py={1} px={6} onSubmit={handleSubmit}>
+      <Heading1 label="Share Your Testimony."></Heading1>
+      <Paragraph1>
+        We want to hear from you. Everyone’s Invited was built on the voices of
+        brave survivors who shared their testimonies with us. If you would like
+        to share your experience please submit it below.
+      </Paragraph1>
+      <Heading3 label="** This is an anonymous submission form **"></Heading3>
+      <Paragraph1>
+        Please do not include names. If names or specific details are included
+        that compromise the anonymity of the testimony, your testimony will not
+        be posted.
+      </Paragraph1>
+      <Heading3 label="Testimony"></Heading3>
+      <TextInput
+        color="primary"
+        bg="secondary"
+        name="testimony"
+        onChange={e => handleChange(e)}
+        value={testimonial.testimony}
+        placeholder="Insert a testimonial..."
+        required
+      />
+      <Heading3 label="School / University / Organisation (Please include if possible)"></Heading3>
```

Comment:
a `label` prop for a heading tag sounds a bit odd, maybe this came from the button, in my mind a label for a button makes sense but for a H1/2/3... maybe not